### DISCUSSION TITLE
MDX V2 and gatsby-source-contentful: Can both work together?

### DISCUSSION THREAD

**Discussion by pelr-meireles** [gatsbyjs/gatsby#36467] [Help]
Hello all, how are you? 
Hope this message find all well! 

I'm currently working on a private project that uses the contentful + MDX (on older version, listed below), and the template pages for rendering the markdown content of contentful pages works fine with the contentful as source provider; There are others plugins of gatsby added to the project (such as the gatsby-source-filesystem, on version 7.5.0; gatsby-source-contentful, on version 7.5.0; and gatsby-plugin-mdx, on version 3.1.0, to list a few).

Recently I'm attempting to upgrade the gatsby extensions, so the project can keep up-to date. When I was attempting to upgrade the  gatsby-plugin-mdx (from version 3.10.1  to the v4 version), the project stopped working as expected, even following the [guidelines to migration of versions](https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/#migrating-from-v3-to-v4). Those pages, on the GraphQL, cannot consult the field childMdx, and if I attempt to use the 'markdown' field, they don't render the page as the previous version.

According to the [documentation of migration, on the breaking changes subtitle](https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/#v3-to-v4-breaking-changes), the plugin only applies for the local files (sourced with the gatsby-source-filesystem), and according to this statement, it does make sense that other sources (such as contentful) doesn't apply...

... The doubt itself is: It is currently possible to use the latest version of the plugin gatsby-plugin-mdx with the gatsby-source-contentful? Or it's only currently possible with the older versions?

If is possible: could an example or reference be provided? I've attempted to research examples of using both plugins (MDX and contentful), but the ones that I found are using older versions of the plugins (which are the current implementation of the project, and it works, but don't accomplish the version upgrade)

If is not: using other gatsby-source-* extensions are on the possible roadmaps for improving the gatsby-plugin-mdx as a solution? Or there are a possibility that all the other gatsby-sources extensions to be mapped/converted to the local file system, so the gatsby-source-filesystem can be used along the mdx extension? (I think that option fulfills the breaking change, and would work for projects that are static, and can be build with the content as static mdx files, for instance)

If there are any other information I can provide, I'm more than glad to share more information. I just cannot share the project itself.

Thanks in advance for the time reading and the patience, and also for the awesome work all the project collaborators do on the gatsby and it's plugins!

**Comment by wahidshafique** [comment]
I have the same issue, with `childMdx` not appearing anywhere

**Comment by wahidshafique** [comment]
I found a Japanese article expanding on this issue https://xenexe.info/597f0d3c-0480-5cc6-b42e-65496c33d093

> According to gatsby-plugin-mdx official document2 , gatsby [-](https://xenexe.info/597f0d3c-0480-5cc6-b42e-65496c33d093#fn-2) plugin-mdx Ver.4.XX cannot read MDX other than file system.
Please Note: Loading MDX from other sources as the file system is not yet supported in gatsby-plugin-mdx@^4.0.0.

The articles 'solution' is just to downgrade.

**Reply by axe312ger** [reply]
Unfortunately, for the time being, this is correct.

MDX v2 can only be server from file system at the moment.

**Reply by wahidshafique** [reply]
@axe312ger Are you aware of any plans to rectify that (genuinely curious, as I haven't really found this issue being discussed elsewhere). I guess given this change, it basically renders many packages that rely on this functionality obsolete in newer versions of Gatbsy right? For example I rely on https://www.gatsbyjs.com/plugins/gatsby-source-airtable/

**Reply by axe312ger** [reply]
As long we have ESM issues (packages only available in ESM; but our toolchain not 100% ESM compatible), this won't be solved. Or only with super ugly hacks.

I tried to ugly hack it, and didn't succeed.

Means I'll wait now for full ESM support, and enjoy all bugs of MDX v1 in my projects :D
